services/web/client/source/class/qxapp/component/filter/TextFilter.js

```diff
@@ -0,0 +1,86 @@
+/* ************************************************************************
+
+   qxapp - the simcore frontend
+
+   https://osparc.io
+
+   Copyright:
+     2019 IT'IS Foundation, https://itis.swiss
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+
+   Authors:
+     * Ignacio Pascual (ignapas)
+
+************************************************************************ */
+
+/**
+ * TextField to make a simple text filter.
+ */
+qx.Class.define("qxapp.component.filter.TextFilter", {
+  extend: qxapp.component.filter.UIFilter,
+
+  /**
+   * Constructor for the TextFilter takes UIFilters mandatory params plus an optional translation id for its label.
+   *
+   * @extends qxapp.component.filter.UIFilter
+   *
+   * @param {string} [labelTr=Filter] Translation id used for the filter label.
+   */
+  construct: function(filterId, groupId, labelTr = "Filter") {
+    this.base(arguments, filterId, groupId);
+    this._setLayout(new qx.ui.layout.Canvas());
+
+    this.__textField = this.getChildControl("textfield").set({
+      placeholder: this.tr(labelTr)
+    });
+
+    this.getChildControl("clearbutton");
+
+    this.__attachEventHandlers();
+  },
+
+  properties: {
+    appearance: {
+      refine: true,
+      init: "textfilter"
+    }
+  },
+
+  members: {
+    __textField: null,
+
+    reset: function() {
+      this.__textField.resetValue();
+      this.__textField.fireDataEvent("input", "");
+    },
+
+    _createChildControlImpl: function(id) {
+      let control;
+      switch (id) {
+        case "textfield":
```

Comment:
this relies on the order the child controls are created to end up with the correct layout ... Using a grid layout would remove this restriction

Reply:
If I change the order, the clear button gets render below the input and is not clickable. But I need it to render on top of the text field...

Reply:
Using a grid would't allow me to do that, right?

Reply:
if you want to super impose buttons you have to use canvas ... and z-index